spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Dilation2DBackpropFilter.scala

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.{DoubleType, FloatType, Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.reflect.ClassTag
+
+class Dilation2DBackpropFilter[T: ClassTag, D: ClassTag](
+         strides: Seq[Int],
+         rates: Seq[Int],
+         padding: String)(implicit ev: TensorNumeric[T], ev2: TensorNumeric[D])
+  extends Operation[Table, Tensor[D], T]{
+
+  private def getOutputSize(inputSize: Int, filterSize: Int, stride: Int, padding: String) = {
```

Comment:
can we add this function to a util class, so dialation and dialtionbackprop can share this?